### PR TITLE
Julia 0.6 fixes

### LINKED ISSUE #26: Warnings on Julia v0.6
>WARNING: deprecated syntax "abstract Transformation" at /home/juliohm/.julia/v0.6/CoordinateTransformations/src/core.jl:16.
Use "abstract type Transformation end" instead.

>WARNING: deprecated syntax "abstract AbstractAffineMap<:Transformation" at /home/juliohm/.julia/v0.6/CoordinateTransformations/src/affine.jl:2.
Use "abstract type AbstractAffineMap<:Transformation end" instead.

### PR DESCRIPTION
**Pull request by c42f**
Some minor fixes for julia-0.6 deprecations and addition of \circ to Base.

Fixes #26

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,4 @@
 julia 0.5
+Compat 0.11.0
```
Comment (yuyichao) [inline REQUIRE]: 0.17.0 for `@compat abstract type`.

**Comment by yuyichao** [comment]
Ping.

File: REQUIRE (hunk 1 above)
Comment (c42f) [inline REQUIRE]: Ah, good point.  I versioned this for `∘`, but forgot about `abstract type`.

**Comment by timholy** [comment]
Would be great to have a new tag for this!

**Comment by c42f** [comment]
Tagged :-)

https://github.com/JuliaLang/METADATA.jl/pull/8898

**Pull request merged**